### PR TITLE
feat(appset): added topic filter for Gitlab SCM

### PR DESCRIPTION
**Pull request by prune998**
Gitlab projects now have `topics` values that are an array of strings that can be associated to repos, like `tags` (now deprecated).

The PR adds a new Gitlab SCM value named `topic` in the CRD that enable to filter targeted repos using a `topic`.

`topic` is a single value string that is used to construct the Gitlab API request to list repos. Not setting it will keep it empty = matching all repos (other selectors still apply, like subgroup and shared projects).

This allows to `filter first` the repo list, cutting down the costs of grabbing the branches and folder tree. It is particularly beneficial in a large org with many repos/sub-folders.

### CODE REVIEW

**Review by ishitasequeira — CHANGES_REQUESTED** [review]
Haven't tested the PR yet but have left some comments.

File: applicationset/services/scm_provider/gitlab_test.go [hunk 1]
```diff
@@ -15,11 +15,12 @@ import (
 
 func gitlabMockHandler(t *testing.T) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
+		t.Log(r.RequestURI)
```
[the remark this answers is not in the document]
Comment (prune998) [reply applicationset/services/scm_provider/gitlab_test.go:18]: done

File: assets/swagger.json [hunk 2]
```diff
@@ -8693,6 +8693,10 @@
         },
         "tokenRef": {
           "$ref": "#/definitions/v1alpha1SecretRef"
+        },
+        "topic": {
+          "description": "Filter repos list based on Gitlab Topic (tag).",
```
[the remark this answers is not in the document]
Comment (prune998) [reply assets/swagger.json:8698]: the `tag` I'm reffering to is actually `tag_list` from https://docs.gitlab.com/ee/api/projects.html#removals-in-api-v5. 
Maybe it's not worth mentioning it at all...

**Comment by prune998** [comment]
> Haven't tested the PR yet but have left some comments.

Thanks for the review. Code updated to resolve your comments.

**Comment by prune998** [comment]
I guess everything is fine now @ishitasequeira ?

**Review by ishitasequeira — CHANGES_REQUESTED** [review]

**Comment by prune998** [comment]
all done @ishitasequeira, sorry I forgot those...

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Comment by prune998** [comment]
thanks @ishitasequeira !

**Pull request merged**